docs/framework/30scoring/implementation.md

```diff
@@ -1,51 +1,187 @@
+| I1012        | Has any version of the on-chain components of the system been audited by more than one audit organisation? If yes, score 0. If no, score 2. |
+| I1013        | For the most recent audit, were all critical issues addressed? If yes, score 0. If no, score 10. |
+| I1014        | Has the deployed version of the on-chain components of the system been audited? If yes, score 0. If no, score 4. |
+| I1015        | Has any version of the off-chain components of the system been audited? If yes, score 0. If no, score 2. If the system contains no off-chain components, then score 0. |
 
 Rationale for scoring:
 
 * Unaddressed critical issues could indicate vulnerabilites.
 * Having an audit, addressing issues, and then deploying the revised code without re-auditing can result in vulnerabilities.
 * The quality of audit firms is not factored in the score as this is very subjective. 
 
+
+
+#### Open Source Code Risk Score
+The risks of not using open source code for a project are discussed in the [Open Source Code](../20categories/30implementation/open-source.md) section.
+
+
+This is currently not accounted for and should be scored as 0.
+
+#### Verified Code on Block Explorer Risk Score
+The risks associated with not publishing code to blockchain explorers is covered in the 
+[Verified Code on Block Explorer](../20categories/30implementation/verified-code.md) section.
+
+
+This is currently not accounted for and should be scored as 0.
+
+
+#### Documentation Risk Score
+The risks associated with not documenting a project well are discussed in the [Documentation](../20categories/30implementation/documentation.md) section.
+
+This is currently not accounted for and should be scored as 0.
+
+
+#### Bug Bounty Risk Score
+Bug Bounties are described in detail in the [Bug Bounty](../20categories/30implementation/bug-bounty.md) section. The equation for the *Bug Bounty Score* is:
+
+```
+Bug Bounty Score = I1401
+```
+
+
+|Question ID  | Question                     |
+|-------------|------------------------------|
+| I1401        | Is there a bug bounty for the project?  How big is the bug bounty? <ul><li>There is no bug bounty: Score 10.</li><li>Less than US$10,000: Score 5.</li><li>US$10,000 but less than US$100,000: Score 2.</li><li>More than US$100,000: Score 0.</li></ul> |
+
+
+
+### Responding to Materialized Risk Score
+```
+Responding to Materialized Risk Score = 
+ Ability to Pause Risk Score + 
+ Ability to Ban Addresses Risk Score
+```
+
+#### Ability to Pause Risk Score
+The ability to pause a project is described in the [Ability to Pause Project](../20categories/30implementation/pause.md) section.
+
+
+This is currently not accounted for and should be scored as 0.
+
+
+#### Ability to Ban Addresses Risk Score
+The ability to ban specific addresses is described in the [Ability to Ban Addresses](../20categories/30implementation/ban-address.md) section.
+
+This is currently not accounted for and should be scored as 0.
```

Comment:
This capability is in some ways antithetical to the value of censorship resistance that most in this space care about. While I think discussing it as a potential concern to be aware of is useful, I would suggest we omit it from any risk scoring and not downrank protocols because they leave censorship decisions to higher layers.

Reply:
I have left the section in the document, but updated it to indicate that concept of not judging the decision.